marquez_airflow/utils.py

```diff
@@ -125,10 +125,17 @@ def get_job_name(task):
     return f'{task.dag_id}.{task.task_id}'
 
 
-def add_airflow_info_to(step_metadata):
-    step_metadata = step_metadata[0]
+def add_airflow_info_to(task, step_metadata):
+    # Add operator info
+    operator = \
+        f'{task.__class__.__module__}.{task.__class__.__name__}'
 
-    step_metadata.context['airflow.version'] = airflow_version
-    step_metadata.context['marquez_airflow.version'] = VERSION
+    step_metadata[0].context['airflow.operator'] = operator
+    step_metadata[0].context['airflow.task_info'] = str(task.__dict__)
 
-    return [step_metadata]
+    # Add version info
+    step_metadata[0].context['airflow.version'] = airflow_version
+    step_metadata[0].context['marquez_airflow.version'] = \
+        MARQUEZ_AIRFLOW_VERSION
```

Comment:
If the current contract is to have a list of StepMetadata, I think we should just loop through them and do this for each (I know right now it's only one but we should bake that assumption in as few places as possible)

Reply:
The fact that it's an array makes working with the extractor a bit cumbersome (given it will always be a single element). But, agreed, let's avoid making this assumption in other parts of the code